### PR TITLE
RPM packaging 4599

### PR DESCRIPTION
**Pull request by ikapelyukhin**

### CODE REVIEW

File: package/rmt.spec [hunk 1]
```diff
@@ -66,32 +66,45 @@ cp -ar . %{buildroot}%{www_base}
...
 rm -rf %{buildroot}%{www_base}/vendor/bundle/ruby/*/gems/*/doc
 rm -rf %{buildroot}%{www_base}/vendor/bundle/ruby/*/gems/*/examples
 rm -rf %{buildroot}%{www_base}/vendor/bundle/ruby/*/gems/*/samples
 rm -rf %{buildroot}%{www_base}/vendor/bundle/ruby/*/gems/*/test
 rm -rf %{buildroot}%{www_base}/vendor/bundle/ruby/*/gems/*/ports
 rm -rf %{buildroot}%{www_base}/vendor/bundle/ruby/*/gems/*/ext
+rm -rf %{buildroot}%{www_base}/vendor/bundle/ruby/*/gems/*/bin
+rm -rf %{buildroot}%{www_base}/vendor/bundle/ruby/*/gems/*/spec
+rm -rf %{buildroot}%{www_base}/vendor/bundle/ruby/*/gems/*/.gitignore
+ls -l %{buildroot}%{www_base}/vendor/bundle/ruby/*/gems/unicode-display_width-1.3.0/data/
```
Comment (ikapelyukhin) [inline package/rmt.spec]: @digitaltom `ls -l`?

**Comment by suse-tests-pass** [comment]
@lagartoflojo will review your pull request :dancers: check https://trello.com/c/SedAyk28/4599-8-rmt-package-it
  @lagartoflojo: Please review this pull request using our guidelines:
  * test for acceptance criteria / functionality
  * check if the new code is covered with tests
  * check for unintended consequences
  * encourage usage of the boyscout rule
  * make sure the code is architected in the best way
  * check that no unnecessary technical debt got introduced
  * make sure that no unnecessary FIXMEs or TODOs got added

File: README.md [hunk 2]
```diff
@@ -4,6 +4,30 @@
 This tool allows you to mirror RPM repositories in your own private network.
 Organization (mirroring) credentials are required to mirror SUSE repositories.
 
+## Installation and configuration
+
+RMT currently gets build for these distributions: `SLE_12_SP2`, `SLE_12_SP3`, `openSUSE_Leap_42.2`, `openSUSE_Leap_42.3`, `openSUSE_Tumbleweed`.
+To add the repository, please call: (replace `<dist>` with your distribution)
```
Comment (lagartoflojo) [inline README.md]: Please remove all `please`s from this and all other files :)
Doc team told us we should not use that word.

File: README.md [hunk 3]
```diff
@@ -4,6 +4,30 @@
...
+
+`zypper ar -f https://download.opensuse.org/repositories/systemsmanagement:/SCC:/RMT/<dist>/systemsmanagement:SCC:RMT.repo`
+
+Please note that on SLES 12 and openSUSE Leap 42.2 you will need to add another repository which provides ruby 2.4, like:
+`https://download.opensuse.org/repositories/OBS:/Server:/Unstable/SLE_12_SP3/OBS:Server:Unstable.repo`
+
+To install rmt, please run: `zypper in rmt`
```
Comment (lagartoflojo) [inline README.md]: s/rmt/RMT/

File: README.md [hunk 4]
```diff
@@ -4,6 +4,30 @@
...
+After installation please configure your RMT instance:
+
+* Create a MySQL/MariaDB user with the following command:
+```
+mysql -u root -p <<EOFF
+GRANT ALL PRIVILEGES ON \`rmt\`.* TO rmt@localhost IDENTIFIED BY 'rmt';
+FLUSH PRIVILEGES;
+EOFF
+```
+* Add your DB and SUSE Customer Center credentials to `database` and `scc` sections respectively of `/etc/rmt.conf` config file
```
Comment (lagartoflojo) [inline README.md]: Which "SUSE Customer Center credentials"?

File: README.md [hunk 5]
```diff
@@ -21,11 +45,20 @@ Organization (mirroring) credentials are required to mirror SUSE repositories.
...
+
 ## Configuration
 
 Available configuration options can be found in `config/rmt.yml` file.
 
-##### Mirroring settings
+### Mirroring settings
 
 - `mirroring.mirror_src` - whether to mirror source (arch = `src`) repos or not.
 - `mirroring.base_dir` - a directory where mirrored files will be stored. HTTP server should be configured to serve files from this directory under `mirroring.mirror_url_prefix`.
```
Comment (lagartoflojo) [inline README.md]: **The** HTTP server....

File: README.md [hunk 6]
```diff
@@ -59,7 +92,7 @@ And accessible at the following URL:
     * `digest_ie`
     * `ntlm_wb`
 
-##### SCC settings for accessing SUSE repositories
+### SCC settings for accessing SUSE repositories
 
 The `scc` section contains your organization credentials for mirroring SUSE repositories.
 Your organization credentials can be obtained at [SUSE Customer Center](https://scc.suse.com).
```
Comment (lagartoflojo) [inline README.md]: obtained **from the** SUSE Customer Center...
Maybe point out where in the SCC the credentials are?

File: README.md [hunk 7]
```diff
@@ -72,20 +105,49 @@ Supported Ruby versions are 2.4.1 and newer.
...
+
+    database_test:
+      <<: *database
+      database: rmt_test
+    ```
 * Run `rails server` to run the web-server
 
+### Packaging
+
+The package is build in the OBS at: https://build.opensuse.org/package/show/systemsmanagement:SCC:RMT/rmt
```
Comment (lagartoflojo) [inline README.md]: is **built**

**Comment by lagartoflojo** [comment]
After (re)installing the package, I get:

```
(9/9) Installing: rmt-0.0.1-28.1.x86_64 ........................................................................................................[done]
Additional rpm output:
groupadd: group 'nginx' already exists
useradd: user '_rmt' already exists
Rails Error: Unable to access log file. Please ensure that /srv/www/rmt/log/production.log exists and is writable (ie, make it writable for user and group: chmod 0664 /srv/www/rmt/log/production.log). The log level has been raised to WARN and the output directed to STDERR until the problem is fixed.
Creating scope :test. Overwriting existing method Subscription.test.
```

Not sure if this is good or bad.

**Comment by lagartoflojo** [comment]
After the above installation:

```
hernan@herninator:~> rmt-cli mirror
Ignoring byebug-8.2.1 because its extensions are not built.  Try: gem pristine byebug --version 8.2.1
Ignoring eventmachine-1.0.8 because its extensions are not built.  Try: gem pristine eventmachine --version 1.0.8
Ignoring executable-hooks-1.3.2 because its extensions are not built.  Try: gem pristine executable-hooks --version 1.3.2
Ignoring gem-wrappers-1.2.7 because its extensions are not built.  Try: gem pristine gem-wrappers --version 1.2.7
Ignoring http_parser.rb-0.6.0 because its extensions are not built.  Try: gem pristine http_parser.rb --version 0.6.0
Ignoring unf_ext-0.0.7.1 because its extensions are not built.  Try: gem pristine unf_ext --version 0.0.7.1
/usr/lib64/ruby/2.4.0/rubygems/core_ext/kernel_require.rb:55:in `require': cannot load such file -- bundler/setup (LoadError)
	from /usr/lib64/ruby/2.4.0/rubygems/core_ext/kernel_require.rb:55:in `require'
	from /srv/www/rmt/config/boot.rb:3:in `<top (required)>'
	from /usr/bin/rmt-cli:13:in `require_relative'
	from /usr/bin/rmt-cli:13:in `<main>'
hernan@herninator:~> sudo rmt-cli mirror
/srv/www/rmt/vendor/bundle/ruby/2.4.0/gems/mysql2-0.4.9/lib/mysql2/client.rb:89:in `connect': Can't connect to local MySQL server through socket '/var/run/mysql/mysql.sock' (2 "No such file or directory") (Mysql2::Error)
	from /srv/www/rmt/vendor/bundle/ruby/2.4.0/gems/mysql2-0.4.9/lib/mysql2/client.rb:89:in `initialize'
	from /srv/www/rmt/vendor/bundle/ruby/2.4.0/gems/activerecord-5.1.3/lib/active_record/connection_adapters/mysql2_adapter.rb:21:in `new'
	from /srv/www/rmt/vendor/bundle/ruby/2.4.0/gems/activerecord-5.1.3/lib/active_record/connection_adapters/mysql2_adapter.rb:21:in `mysql2_connection'
	from /srv/www/rmt/vendor/bundle/ruby/2.4.0/gems/activerecord-5.1.3/lib/active_record/connection_adapters/abstract/connection_pool.rb:759:in `new_connection'
	from /srv/www/rmt/vendor/bundle/ruby/2.4.0/gems/activerecord-5.1.3/lib/active_record/connection_adapters/abstract/connection_pool.rb:803:in `checkout_new_connection'
	from /srv/www/rmt/vendor/bundle/ruby/2.4.0/gems/activerecord-5.1.3/lib/active_record/connection_adapters/abstract/connection_pool.rb:782:in `try_to_checkout_new_connection'
	from /srv/www/rmt/vendor/bundle/ruby/2.4.0/gems/activerecord-5.1.3/lib/active_record/connection_adapters/abstract/connection_pool.rb:743:in `acquire_connection'
	from /srv/www/rmt/vendor/bundle/ruby/2.4.0/gems/activerecord-5.1.3/lib/active_record/connection_adapters/abstract/connection_pool.rb:500:in `checkout'
	from /srv/www/rmt/vendor/bundle/ruby/2.4.0/gems/activerecord-5.1.3/lib/active_record/connection_adapters/abstract/connection_pool.rb:374:in `connection'
	from /srv/www/rmt/vendor/bundle/ruby/2.4.0/gems/activerecord-5.1.3/lib/active_record/connection_adapters/abstract/connection_pool.rb:931:in `retrieve_connection'
	from /srv/www/rmt/vendor/bundle/ruby/2.4.0/gems/activerecord-5.1.3/lib/active_record/connection_handling.rb:116:in `retrieve_connection'
	from /srv/www/rmt/vendor/bundle/ruby/2.4.0/gems/activerecord-5.1.3/lib/active_record/connection_handling.rb:88:in `connection'
	from /srv/www/rmt/vendor/bundle/ruby/2.4.0/gems/activerecord-5.1.3/lib/active_record/model_schema.rb:471:in `load_schema!'
	from /srv/www/rmt/vendor/bundle/ruby/2.4.0/gems/activerecord-5.1.3/lib/active_record/attributes.rb:233:in `load_schema!'
	from /srv/www/rmt/vendor/bundle/ruby/2.4.0/gems/activerecord-5.1.3/lib/active_record/attribute_decorators.rb:50:in `load_schema!'
	from /srv/www/rmt/vendor/bundle/ruby/2.4.0/gems/activerecord-5.1.3/lib/active_record/model_schema.rb:464:in `block in load_schema'
	from /usr/lib64/ruby/2.4.0/monitor.rb:214:in `mon_synchronize'
	from /srv/www/rmt/vendor/bundle/ruby/2.4.0/gems/activerecord-5.1.3/lib/active_record/model_schema.rb:461:in `load_schema'
	from /srv/www/rmt/vendor/bundle/ruby/2.4.0/gems/activerecord-5.1.3/lib/active_record/model_schema.rb:353:in `attribute_types'
	from /srv/www/rmt/vendor/bundle/ruby/2.4.0/gems/activerecord-5.1.3/lib/active_record/model_schema.rb:376:in `type_for_attribute'
	from /srv/www/rmt/vendor/bundle/ruby/2.4.0/gems/activerecord-5.1.3/lib/active_record/table_metadata.rb:32:in `type'
	from /srv/www/rmt/vendor/bundle/ruby/2.4.0/gems/activerecord-5.1.3/lib/active_record/relation/predicate_builder.rb:168:in `build_bind_param'
	from /srv/www/rmt/vendor/bundle/ruby/2.4.0/gems/activerecord-5.1.3/lib/active_record/relation/predicate_builder.rb:114:in `block in create_binds_for_hash'
	from /srv/www/rmt/vendor/bundle/ruby/2.4.0/gems/activerecord-5.1.3/lib/active_record/relation/predicate_builder.rb:89:in `each'
	from /srv/www/rmt/vendor/bundle/ruby/2.4.0/gems/activerecord-5.1.3/lib/active_record/relation/predicate_builder.rb:89:in `create_binds_for_hash'
	from /srv/www/rmt/vendor/bundle/ruby/2.4.0/gems/activerecord-5.1.3/lib/active_record/relation/predicate_builder.rb:34:in `create_binds'
	from /srv/www/rmt/vendor/bundle/ruby/2.4.0/gems/activerecord-5.1.3/lib/active_record/relation/where_clause_factory.rb:21:in `build'
	from /srv/www/rmt/vendor/bundle/ruby/2.4.0/gems/activerecord-5.1.3/lib/active_record/relation/query_methods.rb:612:in `where!'
	from /srv/www/rmt/vendor/bundle/ruby/2.4.0/gems/activerecord-5.1.3/lib/active_record/relation/query_methods.rb:605:in `where'
	from /srv/www/rmt/vendor/bundle/ruby/2.4.0/gems/activerecord-5.1.3/lib/active_record/querying.rb:10:in `where'
	from /srv/www/rmt/lib/rmt/cli/mirror.rb:13:in `mirror'
	from /srv/www/rmt/lib/rmt/cli/main.rb:28:in `block in mirror'
	from /srv/www/rmt/lib/rmt/cli/base.rb:35:in `handle_exceptions'
	from /srv/www/rmt/lib/rmt/cli/main.rb:28:in `mirror'
	from /srv/www/rmt/vendor/bundle/ruby/2.4.0/gems/thor-0.20.0/lib/thor/command.rb:27:in `run'
	from /srv/www/rmt/vendor/bundle/ruby/2.4.0/gems/thor-0.20.0/lib/thor/invocation.rb:126:in `invoke_command'
	from /srv/www/rmt/vendor/bundle/ruby/2.4.0/gems/thor-0.20.0/lib/thor.rb:387:in `dispatch'
	from /srv/www/rmt/lib/rmt/cli/base.rb:24:in `block in dispatch'
	from /srv/www/rmt/lib/rmt/cli/base.rb:35:in `handle_exceptions'
	from /srv/www/rmt/lib/rmt/cli/base.rb:24:in `dispatch'
	from /srv/www/rmt/vendor/bundle/ruby/2.4.0/gems/thor-0.20.0/lib/thor/base.rb:466:in `start'
	from /usr/bin/rmt-cli:34:in `<main>'
```

File: README.md (hunk 7 above)
Comment (lagartoflojo) [reply README.md]: This one is still there

**Comment by digitaltom** [comment]
@lagartoflojo please check the latest package, we have a proper message if the database is not running now. (0.0.1-30.1)

**Pull request merged**